stream-chat-android-client/src/test/java/io/getstream/chat/android/client/socket/SocketFactoryTest.kt

```diff
@@ -73,32 +88,38 @@ internal class SocketFactoryTest {
             randomUser().let {
                 Arguments.of(
                     SocketFactory.ConnectionConf.UserConnectionConf(endpoint, apiKey, it),
-                    "${endpoint}connect?json=${buildFullUserJson(it)}&api_key=$apiKey&authorization=$token&stream-auth-type=jwt"
+                    "${endpoint}connect?json=${buildFullUserJson(it, it.id)}&api_key=$apiKey&authorization=$token&stream-auth-type=jwt"
                 )
             },
             randomUser().let {
                 Arguments.of(
                     SocketFactory.ConnectionConf.UserConnectionConf(endpoint, apiKey, it).asReconnectionConf(),
-                    "${endpoint}connect?json=${buildMinimumUserJson(it)}&api_key=$apiKey&authorization=$token&stream-auth-type=jwt"
+                    "${endpoint}connect?json=${buildMinimumUserJson(it.id)}&api_key=$apiKey&authorization=$token&stream-auth-type=jwt"
                 )
             },
             User("anon").let {
                 Arguments.of(
                     SocketFactory.ConnectionConf.AnonymousConnectionConf(endpoint, apiKey, it).asReconnectionConf(),
-                    "${endpoint}connect?json=${buildMinimumUserJson(it)}&api_key=$apiKey&stream-auth-type=anonymous"
+                    "${endpoint}connect?json=${buildMinimumUserJson(it.id)}&api_key=$apiKey&stream-auth-type=anonymous"
+                )
+            },
+            User("!anon").let {
```

Comment:
Why do we need both `anon` and `!anon`? 🤔

Reply:
It is a test, just to be sure whenever we use `!anon` or `anon` as the user id for an anonymous use the URL to connect with the WebSocket is fine